docs: fully revised guide to create rule that checks response to contain property (#602)
Redocly/openapi-cli: docs/guides/response-contains-property.md

:::

## Steps

1. [Identify the type of object to visit](#identify-the-type-of-object-to-visit). The rules follow the visitor pattern and traverse the tree of the OpenAPI definition.
    - The [Response](https://github.com/Redocly/openapi-cli/tree/master/packages/core/src/types) object is valid as you want to verify the property on the response, not on request.
1. Find relationship between the starting object and the property you must evaluate for the decision. Sometimes, it is the same object as above.
1. [Define the logic for the rule.](#define-the-logic)
1. Report the problem.

### Identify the type of object to visit

As the title says, this guide is about a rule that ensures that the response contains one or more properties. As a result, you cannot use `Schema` as the starting point of your rule because it is used in other areas of the specification, such as requests and parameters.

> Instead, you can use the `Response` for the rule because it is specifically applies to the response schema.

Reviewer: ```suggestion
Instead, you can use the `Response` for the rule because it exclusively applies to the response schema.
```

};
```

### Define the logic

> While you could hardcode the properties you want, it is better to allow configuration of these properties from the `.redocly.yaml` file.

Reviewer 2: ```suggestion
While you could hardcode the properties you want, it is better to allow configuration of these properties from the Redocly configuration file.
```

// ...
```

> So the configuration in the `.redocly.yaml` file would look as follows:

Reviewer 2: ```suggestion
So the configuration in the Redocly configuration file would look as follows:
```

});
```

### Putting it all together

You can copy/paste the code below to try it out.

> 1. Make sure your `.redocly.yaml` file is located in the root repository directory.

Reviewer 2: ```suggestion
1. Make sure your Redocly configuration file is located in the root repository directory.
```

1. Inside the root repository directory, create the `plugins` directory.
1. Add the `demo-plugin.js` file to the `plugins` directory.
1. Inside the `plugins` directory, create the `rules` directory.
1. Add the `response-contains-property.js` file to the `rules` directory.

As a result, your repository hierarchy should look as follows:

> ```

Reviewer: ```suggestion
```shell
```
